### PR TITLE
Feat: App sidebar redesigns

### PR DESCRIPTION
**Pull request by theofficialnar**
Updated app sidebars for the following:

- Schema
- Media
- Reports
- Apps

### Schema
Screenshot_20230815_142144

### Media
Screenshot_20230815_142155

### Reports
Screenshot_20230815_142129

### Apps
Screenshot_20230815_142100

### CODE REVIEW

File: src/apps/marketplace/src/app/components/InstallApp.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import npmIcon from "../../../../../../public/images/npmLogo.png";
+import nodejsIcon from "../../../../../../public/images/nodejsLogo.png";
+import googleChromeIcon from "../../../../../../public/images/googleChromeLogo.png";
+
+export const InstallApp = () => {
+  return (
+    <Stack height="100%" justifyContent="center" alignItems="center">
+      <Box component="img" src={zestyRocket} alt="Zesty Rocket logo" mb={3} />
+      <Stack direction="row" gap={1} mb={8}>
+        <Box
```
[the remark this answers is not in the document]
Comment (theofficialnar) [reply src/apps/marketplace/src/app/components/InstallApp.tsx]: Updated

File: src/apps/marketplace/src/app/components/Sidebar.tsx [hunk 2]
```diff
@@ -28,67 +21,41 @@ export const Sidebar = () => {
     }
   }, []);
 
+  const apps: SubMenu[] = useMemo(() => {
+    if (!!installedApps.length) {
```
[the remark this answers is not in the document]
Comment (theofficialnar) [reply src/apps/marketplace/src/app/components/Sidebar.tsx]: Removed

File: src/apps/media/src/app/components/FolderMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+export const FolderMenu: FC<Props> = ({
+  anchorEl,
+  onCloseMenu,
+  id,
+  groupId,
+  title,
+  binId,
+}) => {
+  const [openDialog, setOpenDialog] = useState<Dialogs>(null);
+  const hiddenGroups =
```
[the remark this answers is not in the document]
Comment (theofficialnar) [reply src/apps/media/src/app/components/FolderMenu.tsx:39]: I've actually tried this but it doesn't work. `useLocalStorage` doesn't get the updated data somehow

File: src/apps/media/src/app/components/Sidebar/Folders.tsx [hunk 4]
```diff
@@ -98,325 +97,406 @@ export const Folders = ({ lockedToGroupId }: Props) => {
...
+          binId,
+          path: `/media/folder/${item.id}`,
+        }),
+        hidden: hiddenGroup.includes(item.id),
+        nodeData: item,
+      }));
+  };
+
+  const mappedTree: TreeItemType[] = useMemo(() => {
+    if (!!binGroups?.length) {
```
[the remark this answers is not in the document]
Comment (theofficialnar) [reply src/apps/media/src/app/components/Sidebar/Folders.tsx]: Removed

File: src/apps/media/src/app/components/Sidebar/NewSidebar.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
+import { FC, useEffect, useState, useRef } from "react";
```
[the remark this answers is not in the document]
Comment (theofficialnar) [reply src/apps/media/src/app/components/Sidebar/NewSidebar.tsx]: Renamed to index.tsx

File: src/apps/schema/src/app/components/Sidebar/ModelList.tsx [hunk 6]
```diff
@@ -44,9 +47,41 @@ export const ModelList = ({ title, models, type }: Props) => {
     setAnchorEl(null);
   };
 
+  const mappedModels: TreeItem[] = useMemo(() => {
+    if (!!models.length) {
```
[the remark this answers is not in the document]
Comment (theofficialnar) [reply src/apps/schema/src/app/components/Sidebar/ModelList.tsx]: Removed

File: src/shell/components/AppSidebar/index.tsx [hunk 7]
```diff
@@ -121,63 +140,86 @@ export const AppSideBar = forwardRef<any, PropsWithChildren<Props>>(
...
-                  size="small"
-                  sx={{
-                    px: 1.5,
-                  }}
-                  onChange={(evt) => setKeyword(evt.target.value)}
-                />
-                {!keyword && (
+                        <SvgIcon
+                          component={titleButtonIcon}
+                          sx={{ fontSize: 18 }}
```
[the remark this answers is not in the document]
Comment (theofficialnar) [reply src/shell/components/AppSidebar/index.tsx]: I've changed all instances of fontSize: 18 to small

**Review by zcolah — CHANGES_REQUESTED** [review]
@Nar overall great job like always. 3 bugs/requests. 

1. User should be able to drag and drop files into ecobins in the sidebar. @agalin920 made this so he should be able to give guidance on this to you so you can make this quickly.

2. File uploaded should appear at the start of the list/grid and not at the end of the grid/list

3.   In the Add Folder Modal, please add all instance bins as options in the Select Folder dropdown. 

https://www.loom.com/share/eaad82a60e744d2c9769fa20796b4f0d?from_recorder=1&focus_title=1

**Review by zcolah — CHANGES_REQUESTED** [review]
Please make the Instance Bin get selected when the user clicks on add subfolder to an Instance Bin in the title bar or in the sidebar. 

See Loom: https://www.loom.com/share/204097fac8a645fc88ec8a37bf74c702

**Comment by theofficialnar** [comment]
@zcolah added the following as requested:

- Show all folders and ecobins when clicking the generic + folder button
- Automatically select ecobin when adding a new folder in ecobin root
- Show warning message when attempting to drag a folder to a diff ecobin
- Allow dragging of file to parent ecobin's root

> File uploaded should appear at the start of the list/grid and not at the end of the grid/list
As discussed this has already been resolved elsewhere

File: src/apps/media/src/app/components/Sidebar/index.tsx [hunk 8]
```diff
@@ -1,42 +1,110 @@
...
+      icon: ScheduleRounded,
+    },
+    {
+      name: "Insights",
+      path: "/media/insights",
+      icon: InsightsRounded,
+    },
+  ];
+
+  const renderTitleButton = () => {
```
[the remark this answers is not in the document]
Comment (theofficialnar) [reply src/apps/media/src/app/components/Sidebar/index.tsx]: Gotcha, moved it to its own component

**Review by zcolah — CHANGES_REQUESTED** [review]
Was making a demo video and discovered two bugs which we need to resolve ideally before shipping. 

1. In the create folder experience, remove none as an option from the dropdown, and make the default selection as the first ecobin in an instance
2. Always show the instance ecobin even if it has 0 sub folders 

See video for more details: https://www.loom.com/share/65acb0242c4345f495c2e7656f0ecaf9

**Comment by shrunyan** [comment]
image

app crash when clearing local storage

**Review by zcolah — CHANGES_REQUESTED** [review]
**Problem**
Currently it is not evident to a user what order they have sorted everything by.

CleanShot 2023-08-24 at 04 38 57@2x

**Solution**
Please have the selected menu option be in the selected state so that it is clear to the user what the current sorting is.

**Pull request merged**